cqf-fhir-cr/src/main/java/org/opencds/cqf/fhir/cr/questionnaire/populate/PopulateRequest.java

```diff
@@ -23,38 +23,39 @@ public class PopulateRequest implements IQuestionnaireRequest {
     private final ModelResolver modelResolver;
     private final FhirVersionEnum fhirVersion;
     private final String defaultLibraryUrl;
+    private final Boolean useServerData;
+    private final IInputParameterResolver inputParameterResolver;
     private IBaseOperationOutcome operationOutcome;
 
-    // test constructor
-    public PopulateRequest(FhirVersionEnum fhirVersion, String operationName) {
-        this.operationName = operationName;
-        this.questionnaire = null;
-        this.subjectId = null;
-        this.parameters = null;
-        this.bundle = null;
-        this.libraryEngine = null;
-        this.modelResolver = FhirModelResolverCache.resolverForVersion(fhirVersion);
-        this.fhirVersion = fhirVersion;
-        this.defaultLibraryUrl = null;
-    }
-
     public PopulateRequest(
             String operationName,
             IBaseResource questionnaire,
             IIdType subjectId,
             IBaseParameters parameters,
             IBaseBundle bundle,
+            Boolean useServerData,
             LibraryEngine libraryEngine,
             ModelResolver modelResolver) {
         this.operationName = operationName;
         this.questionnaire = questionnaire;
         this.subjectId = subjectId;
         this.parameters = parameters;
         this.bundle = bundle;
+        this.useServerData = useServerData;
         this.libraryEngine = libraryEngine;
         this.modelResolver = modelResolver;
         this.fhirVersion = questionnaire.getStructureFhirVersionEnum();
         this.defaultLibraryUrl = resolveDefaultLibraryUrl();
+        inputParameterResolver = libraryEngine == null
+                ? null
```

Comment:
question: As far as I can tell, the libraryEngine is a required parameter. The implementation of `getParameters` depends on it existing. Rather than a `null` check, is it possible to use a `notNull` assertion?

Reply:
Something like this?

Reply:
Yes, but was hoping to get both code paths unified. IOW, what's the use case where we supply a `inputParametersResolver` independent of the `libraryEngine`. Can we always just create a `inputParametersResolve` off of Library engine? Or vice-versa?

Reply:
Granted, this is a nitpick, so feel free to say no. Just wondering if we can unify it all.

Reply:
In the PlanDefinition $apply case I allow for a custom resolver to be passed in, but we don't really have a use case for that yet.  Definitely doable but I would want to spend a little more time on it so think it would be best in another PR.